### PR TITLE
[export] 'activity_exception' should not be preselected

### PR DESCRIPTION
**Pull request by ppr-odoo**
Pad:https://pad.odoo.com/p/r.4bc3c0bd86ca00af2b39ebe1a85912c7
Task:https://www.odoo.com/web#id=2075083&action=333&active_id=967&model=project.task&view_type=form&menu_id=4720
Fixes & Closes https://github.com/odoo/odoo/issues/37850

### CODE REVIEW

**Review by LucasLefevre — COMMENTED** [review]
Hello @ppr-odoo 
I think adding an option `preselect_export` is not needed.
The option seems always used with `widget="activity_exception"`.
This makes the option redundant with the widget attribute, hence it is useless.
See the change I propose in my comment.

File: addons/web/static/src/js/views/list/list_controller.js [hunk 1]
```diff
@@ -351,7 +352,12 @@ var ListController = BasicController.extend({
      */
     _getExportDialogWidget() {
         let state = this.model.get(this.handle);
-        let defaultExportFields = this.renderer.columns.filter(field => field.tag === 'field').map(field => field.attrs.name);
+        let defaultExportFields = this.renderer.columns.filter(field => {
+            var options = field.attrs.options ? pyUtils.py_eval(field.attrs.options) : {};
+            if (options.preselect_export !== false && field.tag === 'field') {
+                return field;
+            }
+        }).map(field => field.attrs.name);
```
[the remark this answers is not in the document]
Comment (msh-odoo) [reply addons/web/static/src/js/views/list/list_controller.js]: @LucasLefevre Good catch, will check and do changes accordingly but note that, with this change, activity_exception will never be preselected, if someone wants to keep preselected for specific model, It's OK for me, user can manually add activity_exception field right?

[the remark this answers is not in the document]
Comment (LucasLefevre) [reply addons/web/static/src/js/views/list/list_controller.js]: Indeed, activity_exception will never be preselected. After all it's what the spec says:

> A field with the 'activity_exception' widget should NEVER be preselected for export

And as you say, the field can always be manually added

[the remark this answers is not in the document]
Comment (msh-odoo) [reply addons/web/static/src/js/views/list/list_controller.js]: @LucasLefevre Done 👍

**Comment by tde-banana-odoo** [comment]
Canceled

**Pull request closed without merging**